File: SDKs/googleapis.com/documentai/v1beta2/python/src/sdk/models/shared/googleclouddocumentaiv1beta2processdocumentrequest.py
from dataclasses import dataclass, field
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class GoogleCloudDocumentaiV1beta2ProcessDocumentRequest:
    r"""GoogleCloudDocumentaiV1beta2ProcessDocumentRequest
    Request to process one document.
    """
    
    automl_params: Optional[GoogleCloudDocumentaiV1beta2AutoMlParams] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('automlParams') }})
    document_type: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('documentType') }})
    entity_extraction_params: Optional[GoogleCloudDocumentaiV1beta2EntityExtractionParams] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('entityExtractionParams') }})
    form_extraction_params: Optional[GoogleCloudDocumentaiV1beta2FormExtractionParams] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('formExtractionParams') }})
    input_config: Optional[GoogleCloudDocumentaiV1beta2InputConfig] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('inputConfig') }})
    ocr_params: Optional[GoogleCloudDocumentaiV1beta2OcrParams] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ocrParams') }})
    output_config: Optional[GoogleCloudDocumentaiV1beta2OutputConfig] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('outputConfig') }})
    parent: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('parent') }})
    table_extraction_params: Optional[GoogleCloudDocumentaiV1beta2TableExtractionParams] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('tableExtractionParams') }})
    

File: SDKs/googleapis.com/documentai/v1beta2/typescript/src/sdk/models/shared/googleclouddocumentaiv1beta2processdocumentrequest.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { GoogleCloudDocumentaiV1beta2AutoMlParams } from "./googleclouddocumentaiv1beta2automlparams";
import { GoogleCloudDocumentaiV1beta2EntityExtractionParams } from "./googleclouddocumentaiv1beta2entityextractionparams";
import { GoogleCloudDocumentaiV1beta2FormExtractionParams } from "./googleclouddocumentaiv1beta2formextractionparams";
import { GoogleCloudDocumentaiV1beta2InputConfig } from "./googleclouddocumentaiv1beta2inputconfig";
import { GoogleCloudDocumentaiV1beta2OcrParams } from "./googleclouddocumentaiv1beta2ocrparams";
import { GoogleCloudDocumentaiV1beta2OutputConfig } from "./googleclouddocumentaiv1beta2outputconfig";
import { GoogleCloudDocumentaiV1beta2TableExtractionParams } from "./googleclouddocumentaiv1beta2tableextractionparams";



// GoogleCloudDocumentaiV1beta2ProcessDocumentRequest
/** 
 * Request to process one document.
**/
export class GoogleCloudDocumentaiV1beta2ProcessDocumentRequest extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=automlParams" })
  automlParams?: GoogleCloudDocumentaiV1beta2AutoMlParams;

  @SpeakeasyMetadata({ data: "json, name=documentType" })
  documentType?: string;

  @SpeakeasyMetadata({ data: "json, name=entityExtractionParams" })
  entityExtractionParams?: GoogleCloudDocumentaiV1beta2EntityExtractionParams;

  @SpeakeasyMetadata({ data: "json, name=formExtractionParams" })
  formExtractionParams?: GoogleCloudDocumentaiV1beta2FormExtractionParams;

  @SpeakeasyMetadata({ data: "json, name=inputConfig" })
  inputConfig?: GoogleCloudDocumentaiV1beta2InputConfig;

  @SpeakeasyMetadata({ data: "json, name=ocrParams" })
  ocrParams?: GoogleCloudDocumentaiV1beta2OcrParams;

  @SpeakeasyMetadata({ data: "json, name=outputConfig" })
  outputConfig?: GoogleCloudDocumentaiV1beta2OutputConfig;

  @SpeakeasyMetadata({ data: "json, name=parent" })
  parent?: string;

  @SpeakeasyMetadata({ data: "json, name=tableExtractionParams" })
  tableExtractionParams?: GoogleCloudDocumentaiV1beta2TableExtractionParams;
}
